### DISCUSSION TITLE
Using snake case naming for json serialization

### DISCUSSION THREAD

**Discussion by hgjunip** [Azure/azure-mobile-apps#572] [Q&A]
As the title suggests, in our project we are using snake case naming for Json serialization. We did this by creating our own implementation of the extension method AddDatasyncControllers. Ours is pretty much the same as the one in the library except for the Json serialization settings. 

Most things have been working fine except for incremental sync. This seems to be related to the use of SystemProperties.JsonUpdatedAtProperty which is defined as "updatedAt" so our client never reads the value and in our case syncs always pull all objects. I've tried using the UpdatedAtAttribute in the client but still no delta sync token is saved.

I was wondering if there was a way we could get around this issue? I understand that snake case naming is not supported but any guidance is appreciated.

**Comment by adrianhall** [comment]
You will have to fork the repository and update the client to use snake case.  There are a lot of places where camel case is assumed, so it's a large fix.
